extensions/amp-story/1.0/amp-story.js

```diff
@@ -1284,17 +1328,53 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
-    if (
-      isExperimentOn(this.win, 'amp-story-paywall-exp') &&
-      targetPage.isPaywallProtected()
-    ) {
-      if (this.storeService_.get(StateProperty.SUBSCRIPTIONS_DIALOG_STATE)) {
-        // Subscription dialog is already triggered.
-        return Promise.resolve();
-      }
-      this.storeService_.dispatch(Action.TOGGLE_SUBSCRIPTIONS_DIALOG, true);
+    const subscriptionsState = this.storeService_.get(
+      StateProperty.SUBSCRIPTIONS_STATE
+    );
+    if (this.isPaywallStory_()) {
+      if (
+        pageIndex >= PAYWALL_PAGE_INDEX &&
+        subscriptionsState !== SubscriptionsState.GRANTED
+      ) {
+        if (subscriptionsState === SubscriptionsState.UNKNOWN) {
```

Comment:
This should be `if (subscriptionsState === SubscriptionsState.UNKNOWN && pageIndex >= PAYWALL_PAGE_INDEX)` directly line 1334, there is no reason for it to be nested into three levels of `if`

Coupled with the `SubscriptionsState.NO_SUB` suggested below this will remove the need for other `if`s